src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,176 @@
+    int LECTURE_ID = 1;
+    String CONTENT = "Test content";
+    String EMAIL = "test@example.com";
+    testServices.setEnvIsLoggedIn(true);
+    testServices.setEnvEmail(EMAIL);
+    testServices.setEnvAuthDomain("example.com");
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, Integer.toString(LECTURE_ID));
+    request.setContent(CONTENT);
+    PreparedQuery query = datastore.prepare(new Query(Comment.ENTITY_KIND));
+    assertThat(query.countEntities(withLimit(2))).isEqualTo(0);
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_ACCEPTED);
+    assertThat(query.countEntities(withLimit(2))).isEqualTo(1);
+    Comment comment = Comment.fromEntity(query.asSingleEntity());
+    assertThat(comment.lecture().getId()).isEqualTo(LECTURE_ID);
+    assertThat(comment.author().getEmail()).isEqualTo(EMAIL);
+    assertThat(comment.content()).isEqualTo(CONTENT);
+    assertThat(comment.parent().isPresent()).isFalse();
+  }
+
+  @Test
+  public void doPost_storesCommentParent() throws ServletException, IOException {
+    int PARENT_ID = 32;
+    testServices.setEnvIsLoggedIn(true);
+    testServices.setEnvEmail("test@example.com");
+    testServices.setEnvAuthDomain("example.com");
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "1");
+    request.setParameter(DiscussionServlet.PARAM_PARENT, Integer.toString(PARENT_ID));
+    request.setContent("Test content");
+    PreparedQuery query = datastore.prepare(new Query(Comment.ENTITY_KIND));
+    assertThat(query.countEntities(withLimit(2))).isEqualTo(0);
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_ACCEPTED);
+    Comment comment = Comment.fromEntity(query.asSingleEntity());
+    assertThat(comment.parent().isPresent()).isTrue();
+    assertThat(comment.parent().get().getId()).isEqualTo(PARENT_ID);
+  }
+
+  @Test
+  public void doGet_returnsNothingForUnknownLecture() throws ServletException, IOException {
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "1");
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    assertThat(response.getContentAsString()).startsWith("[]");
+  }
+
+  @Test
+  public void doGet_returnsComments() throws ServletException, IOException {
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "1");
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<Comment>>() {}.getType();
```

Comment:
Up to you, this code is used more than once, maybe make a helper function to clean up the test

Reply:
Created helper function